### PR TITLE
impl(docfx): introduce yaml-cpp dependency

### PR DESCRIPTION
**Pull request by coryan**
I will need to generate YAML files. These look simple until you start reading about how to quote string literals.

Part of the work for #10895

- - -
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/10964)

### CODE REVIEW

File: bazel/google_cloud_cpp_deps.bzl [hunk 1]
```diff
@@ -229,3 +229,13 @@ def google_cloud_cpp_deps(name = None):
         strip_prefix = "pugixml-1.13",
         build_file = Label("//bazel:pugixml.BUILD"),
     )
+
+    maybe(
+        http_archive,
+        name = "com_github_jbeder_yaml_cpp",
+        urls = [
+            "https://github.com/jbeder/yaml-cpp/archive/refs/tags/yaml-cpp-0.7.0.tar.gz",
```
Comment (devbww) [inline bazel/google_cloud_cpp_deps.bzl:237]: Do we want to cache it in GCS?

Comment (devbww) [inline bazel/google_cloud_cpp_deps.bzl:237]: s:refs/tags/:: ?

Comment (coryan) [reply bazel/google_cloud_cpp_deps.bzl]: Done.

Comment (coryan) [reply bazel/google_cloud_cpp_deps.bzl]: Yes.  Funny thing, I had already created the copy and everything.

Comment (devbww) [reply bazel/google_cloud_cpp_deps.bzl]: If you add the alternative URL first, you should just be able to `bazel/deps-cache.py -p`.

**Review by devbww — APPROVED** [review]

**Pull request merged**